amundsen_application/__init__.py

```diff
@@ -56,6 +57,7 @@ def create_app(config_module_class: str, template_folder: str = None) -> Flask:
     app.register_blueprint(announcements_blueprint)
     app.register_blueprint(log_blueprint)
     app.register_blueprint(mail_blueprint)
+    app.register_blueprint(dashboard_blueprint)
```

Comment:
- New blueprint: The decision to make a new file was in some part to get something up and running, but mostly because the `metadata_blueprint` is growing to an unmanageable size. Adding dashboards might double this file, and who knows what additional resources will be added in the future. We could probably separate the table and user APIs.
- As far as code sharing goes, it is possible for an API to handle multiple resources, but I don't think that's the best design choice. I would rather build multiple simple APIs custom built for each case. I think code reuse will largely be limited to utility functions. As for redux/saga code, that's another story we'll have to think about more.
- Table `key` is a field that only exists on the frontend. I think we should use a consistent field for identifying all resources, whether it is a table, dashboard, or user. I have discussed with @jinhyukchang about adding `uri` in the response for all resources, though I'm not sure what the `uri` for a user would be.

Reply:
I see. In regards to an API that handles multiple resources, the current plan for search is to have the filter endpoint serve multiple resources which seems ideal for maintenance, and shouldn't pose a problem if APIs are consistent. I would want to do more investigation into the right way to approach it though, and I think we could pay some more attention to future choices made for the APIs in our other services for scalability and ultimately apply them here. 

For us, if the use case we consider is defined as "get resource metadata" vs. "get table metadata" and "get dashboard metadata" and "get <future-resource> metadata" -- and if we work towards consistent APIs to allow this I believe that is the best choice long term.